src/components/fileStorage/repo/files.repo.js

```diff
@@ -38,7 +44,10 @@ const removePersonalFilesByUserId = async (userId) => {
 	if (!isValidObjectId(userId)) {
 		throw new AssertionError(missingParameters({ userId }));
 	}
-	const deleteResult = await FileModel.deleteMany(personalFileSearchQuery(userId)).lean().exec();
+	const personalNotDeletedFileSearchQuery = {
+		$and: [personalFileSearchQuery(userId), { deletedAt: { $exists: false } }],
```

Comment:
there is also deleted field, which is the one that matters

Reply:
That's a little bit tricky. With this query I want to ensure that already deleted files are not deleted again, because otherwise the deletedAt flag would be updated to the new deletion day and the time until deletion could be extended. If there is a file with deleted true, but without a deletedAt flag (that should not be the case right now, because no files are deleted yet and new deleted files will always get both flags), I think it is OK to update/add the deletedAt flag there. Otherwise I am not sure how we would handle this case in general. Alternatively we can in our deletion script, delete any files right away if they are marked as deleted, but don't have a deletedAt flag?